### DISCUSSION TITLE
Migrate complex business logic from asp net framework to orchard core

### DISCUSSION THREAD

**Discussion by kudryavtsevda** [OrchardCMS/OrchardCore#11421] [Q&A]
I faced with issue that I need to replicate a part of complex business logic from existing legacy asp.net mvc 4.7.2 framework.
So this complex logic contains several pages which iteracts somehow via existing API (search engine, display search results in a list, open a particular item, add opened item into basket and etc).
I want to keep that pages as is in terms of html and keep interaction with API as is BUT I need to be able to change some CSS on that pages. In that case I came up with idea to create orchard module and replicate pages and interactions between and third party API using simple MVC using controllers, views and models because I don't want to provide ability for end-user to edit them expect CSS. 
This CSS editing will be a part of admin menu modules where user can choose colors,background colors and etc.

And I am asking whether my idea to use simple orchard module as simple MVC project is good idea or should I use orchard's content items/shapes/parts/templates? Also I don't understand how to embed this hard-coded pages with business logic into orchard's content items/types. Could you provide some your idea and samples?

Unfortunately I am not so experienced in orchard and I am looking for several options to choose one of them.

**Comment by hishamco** [comment]
You can put your existing app into a module if you want but you need to migrate everything to ASP.NET Core or put your views and controllers directly into the host app, coz at the end host app is ASP.NET Core

**Comment by ns8482e** [comment]
Orchard Core runs on ASP.NET CORE - .NET 6 and not on  asp.net mvc 4.7.2 framework.

**Reply by kudryavtsevda** [reply]
@hishamco @ns8482e I know. I am talking about making from the scratch that templates or shapes in terms of Orchard to achieve the same result and to be able to see exact the same complex process which is converted from asp net framework to asp net core mvc in orchard core

**Reply by Skrypt** [reply]
Yes, everything is possible based on the extent of knowledge you want to acquire on the actual OC framework.

**Reply by kudryavtsevda** [reply]
@Skrypt could you tell what should I do first if I want to create a readonly html document which represents simple search box and button find? I want to use it as a part of whole page

**Reply by Skrypt** [reply]
You want to search for what in that form?
